cylc/flow/suite_db_mgr.py

```diff
@@ -355,19 +355,18 @@ def put_suite_template_vars(self, template_vars):
 
     def put_task_event_timers(self, task_events_mgr):
         """Put statements to update the task_action_timers table."""
-        if task_events_mgr.event_timers:
```

Comment:
This variable, `event_timers` in the task event manager, does not hold task poll and try timers (they are task proxy attributes) which are also written to the `task_action_timers` DB table.  It only holds job log retrieval timers (so suites that need remote job log retrieval presumably do get correct DB housekept correctly - this might explain why no one has noticed this problem in operational suites?).

Reply:
Due to this line:

https://github.com/cylc/cylc-flow/blob/35b160453601fbb7cd92a83e6f3251b90b758b9b/cylc/flow/suite_db_mgr.py#L204

It would appear that there is no consequence to this line: 

```python
self.db_deletes_map[self.TABLE_TASK_ACTION_TIMERS].append({})
```

So we should be good.

Reply:
> It would appear that there is no consequence to this line

Huh, by experiment, this change results in the table been emptied, and then repopulated with only the current timers.

Reply:
Aah so an empty "delete map" means "refresh".